pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -179,6 +179,19 @@ def test_setitem_with_unaligned_sparse_value(self):
         expected = Series(SparseArray([1, 0, 0]), name="new_column")
         tm.assert_series_equal(df["new_column"], expected)
 
+    def test_setitem_period_preserves_dtype(self):
+        # GH: 26861
+        data = [Period("2003-12", "D")]
+        result = DataFrame([])
+        result["a"] = data
+        result_dtype = result["a"].dtype
+
+        expected = DataFrame({"a": data})
+        expected_dtype = PeriodDtype(freq="D")
+
+        assert result_dtype == expected_dtype
```

Comment:
This shouldn't be needed. `tm.assert_frame_equal` will also check the dtypes of `result` and `expected`

Reply:
Thanks, I've removed the dtype check